webrender/src/renderer.rs

```diff
@@ -768,9 +809,23 @@ fn create_prim_shader(name: &'static str,
         VertexFormat::Blur => DESC_BLUR,
     };
 
-    device.create_program(name,
-                          &prefix,
-                          &vertex_descriptor)
+    let program = device.create_program(name,
+                                        &prefix,
+                                        &vertex_descriptor);
+
+    if let Ok(ref program) = program {
+        device.bind_shader_sampler(program, "sColor0", TextureSampler::Color0);
```

Comment:
Not sure what being generic over `Into<TextureSlot>` gives us here. Would it be simpler to just do:
```rust
device.bind_shader_textures(program, &[
  ("sColor0", 0),
  ("sColor1", 1),
  ... // this table could be defined as a static const on top and re-used, if needed, for different programs
]);
```

Reply:
It means that there is only one place in the code that does the mapping from a sampler enum to an integer. It seems like that might make it harder to accidentally get things out of sync - a minor detail, certainly.